src/Nutgram.php

```diff
@@ -253,16 +253,33 @@ public function processUpdate(Update $update): void
         $this->fireHandlers($handlers);
     }
 
+    /**
+     * @param  string  $type
+     * @param  array  $parameters
+     * @return mixed
+     * @throws Throwable
+     */
+    protected function fireHandlersBy(string $type, array $parameters = []): mixed
+    {
+        $handlers = [];
+        $this->addHandlersBy($handlers, $type);
+        return $this->fireHandlers($handlers, $parameters);
+    }
+
     /**
      * @param  array  $handlers
+     * @param  array  $parameters
+     * @return mixed
      * @throws Throwable
      */
-    protected function fireHandlers(array $handlers): void
+    protected function fireHandlers(array $handlers, array $parameters = []): mixed
     {
+        $result = null;
+
         /** @var Handler $handler */
         foreach ($handlers as $handler) {
             try {
-                $handler->getHead()($this);
+                $result = $handler->getHead()($this, ...$parameters);
```

Comment:
The Handler's `__invoke` method doesn't have a 2nd parameter `$parameters` so it's wrong.
You should find a way to fill the `$parameters` property in Handler maybe using `$handler->setParameters($parameters);` before this line